### PR TITLE
BionicFramework Refactor

### PR DESCRIPTION
**Pull request by roshanr10**
This turned into a massive refactor of everything to allow for easier testing… currently everything besides the drivetrain is here, will push that soon..

### CODE REVIEW

File: src/org/team4909/bionicframework/oi/BionicJoystick.java [hunk 1]
```diff
@@ -60,14 +46,29 @@ public void buttonToggled(BionicButton button, Command command){
...
+	
+	private class BionicJoystickAxisButton extends Trigger {
+		private BionicJoystick inputJoystick;
+		private int axisNumber;
+		
+		private double thresholdValue;
+		
+		public BionicJoystickAxisButton(BionicJoystick joystick, int axis, double minThreshold) {
+			inputJoystick = joystick;
+
```
Comment (twinters007) [inline src/org/team4909/bionicframework/oi/BionicJoystick.java]: Leaving room for jesus?

File: src/org/team4909/bionicframework/hardware/PotentiometerController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	public PotentiometerController(SpeedController motor, AnalogPotentiometer potentiometer, double p, double i, double d) {
+		super(p, i, d);
+
+		this.motor = motor;
+		this.potentiometer = potentiometer;
+
+		getPIDController().setContinuous(false);
+		
+		getPIDController().enable();
```
Comment (TechplexEngineer) [inline src/org/team4909/bionicframework/hardware/PotentiometerController.java]: When do you ever .disable() the pid controller?

Comment (roshanr10) [reply src/org/team4909/bionicframework/hardware/PotentiometerController.java]: we don't. it's constantly held at angle.

Comment (roshanr10) [reply src/org/team4909/bionicframework/hardware/PotentiometerController.java]: think we should have a way to easily disable?

**Pull request merged**